Sources/KeyboardSupport/KeyboardRespondable.swift

```diff
@@ -137,33 +137,48 @@ public extension KeyboardScrollable where Self: UIViewController {
     }
     
     private func adjustViewForKeyboardAppearance(with keyboardInfo: KeyboardInfo, firstResponder: UIView) {
+        guard let scrollView = keyboardScrollableScrollView else { return }
+        
         var mutableInset: UIEdgeInsets
-        if shouldPreserveContentInsetWhenKeyboardVisible, let originalContentInset = keyboardScrollableScrollView?.originalContentInset {
+        if shouldPreserveContentInsetWhenKeyboardVisible, let originalContentInset = scrollView.originalContentInset {
             mutableInset = originalContentInset
         } else {
             mutableInset = .zero
         }
         
         // Adjust scroll view insets for keyboard height
-        let keyboardSize = keyboardInfo.finalFrame.size
+        let keyboardHeight = keyboardInfo.finalFrame.height
         if #available(iOS 11.0, *) {
-            mutableInset.bottom += keyboardSize.height - view.safeAreaInsets.bottom
+            mutableInset.bottom += keyboardHeight - view.safeAreaInsets.bottom
         } else {
-            mutableInset.bottom += keyboardSize.height
+            mutableInset.bottom += keyboardHeight
         }
+        
         adjustScrollViewInset(mutableInset)
         
         // If active text field is hidden by keyboard, scroll so it's visible
-        let keyboardMinY = view.bounds.height - keyboardSize.height
-        
-        let firstResponderConvertedFrame = firstResponder.convert(firstResponder.bounds, to: nil)
-        let firstResponderMaxY = firstResponderConvertedFrame.maxY
-        
-        if firstResponderMaxY > keyboardMinY {
-            keyboardScrollableScrollView?.scrollRectToVisible(firstResponder.frame, animated: true)
+        if let textView = firstResponder as? UITextView {
+            // Get the frame of the cursor/selection to improve scrolling position for UITextView's
+            // DispatchQueue.async() is necessary because the selectedTextRange typically hasn't not been updated when UIResponder.keyboardWillShowNotification is posted
+            DispatchQueue.main.async {
+                guard let textRange = textView.selectedTextRange, let selectionRect = textView.selectionRects(for: textRange).first else { return }
+                // Set an arbitray width to the target CGRect in case the width is zero. Otherwise, scrollRectToVisible has no effect.
+                let contentRect = textView.convert(selectionRect.rect, to: scrollView).modifying(width: 30)
```

Comment:
Is there a reason we didn't just set the width to 1? 30 just seems too...well...arbitrary... 🤣 Not a huge deal either way though because of the helpful comment.

Reply:
Clearly because it's 30 times better than 1.

Reply:
Forgot to update before merging. Will change and test in another PR. :D